### PR TITLE
[ST] Big refactor of Test Storage usage in ST

### PR DESCRIPTION
**Pull request by jankalinic**
### Type of change

- Refactoring

### Description

This PR should consist of all the planned changes for testStorage class and its usage throughout the STs. The main focus point is the replacement of redundant variables that were being set directly from test storage, now there are only those with specific meaning to them. This PR also includes the changes to initialization of the TestStorage, every test should get it via the extensionContext from StorageMap.

### Checklist

- [ ] Write tests
- [ ] Make sure all tests pass
- [ ] Check RBAC rights for Kubernetes / OpenShift roles
- [ ] Try your changes from Pod inside your Kubernetes and OpenShift cluster, not just locally

### CODE REVIEW

**Comment by jankalinic** [comment]
@strimzi-ci run tests --cluster-type=ocp --cluster-version=4.14 --install-type=bundle --profile=acceptance

File: systemtest/src/test/java/io/strimzi/systemtest/bridge/HttpBridgeST.java [hunk 1]
```diff
@@ -67,19 +66,14 @@ class HttpBridgeST extends AbstractST {
...
 
-    private String kafkaClientsPodName;
-
     @ParallelTest
     void testSendSimpleMessage(ExtensionContext extensionContext) {
-        final TestStorage testStorage = new TestStorage(extensionContext);
-        final String topicName = testStorage.getTopicName();
-        final String producerName = "producer-" + new Random().nextInt(Integer.MAX_VALUE);
-        final String consumerName = "consumer-" + new Random().nextInt(Integer.MAX_VALUE);
+        final TestStorage testStorage = storageMap.get(extensionContext);
```
Comment (im-konge) [inline systemtest/src/test/java/io/strimzi/systemtest/bridge/HttpBridgeST.java:71]: You have it all places, but why it's better to use `storageMap` than `new TestStorage()`? I think that ultimately you should not have access to `storageMap` directly in the tests and you should have everything generated based on the `extenstionContext`. IIRC we wanted to remove `storageMap` and use `new TestStorage`.

**Review by henryZrncik — APPROVED** [review]
yeah as Lukas mentioned, in the discussion we agreed to go with having all data from `extensionContext` or/and going with `TestStorage`, but apart from this I do believe that rest of refactor is as we spoke about so LGTM so far.  
Also should we not run more than acceptance profile on changes in basically almost all of the test classes ?

File: systemtest/src/test/java/io/strimzi/systemtest/bridge/HttpBridgeST.java (hunk 1 above)
Comment (jankalinic) [reply systemtest/src/test/java/io/strimzi/systemtest/bridge/HttpBridgeST.java:71]: I based this upon the storageMap logic, as its a beforeEach thing, where the testStorage instance is being put into the map, but if you say there will be a storageMap removal soon, then these lines of this refactor make no sense. I should rewrite these lines to be rather new obj init instead. Thanks !

**Comment by jankalinic** [comment]
> yeah as Lukas mentioned, in the discussion we agreed to go with having all data from `extensionContext` or/and going with `TestStorage`, but apart from this I do believe that rest of refactor is as we spoke about so LGTM so far. Also should we not run more than acceptance profile on changes in basically almost all of the test classes ?

Based on the character of this refactor, i took the acceptance testrun as a minimum pass indicator, you are correct as all tests should be run.

**Review by see-quick — APPROVED** [review]
LGTM, assuming that tests pass.

File: systemtest/src/test/java/io/strimzi/systemtest/log/LogSettingST.java [hunk 2]
```diff
@@ -276,9 +272,8 @@ void testKafkaLogSetting(ExtensionContext extensionContext) {
     @ParallelTest
     @Tag(CONNECT)
     void testConnectLogSetting(ExtensionContext extensionContext) {
-        final TestStorage testStorage = storageMap.get(extensionContext);
-        String clusterName = testStorage.getClusterName();
-        String connectClusterName = clusterName + "-connect";
+        final TestStorage testStorage = new TestStorage(extensionContext);
+        String connectClusterName = testStorage.getClusterName() + "-connect";
```
Comment (im-konge) [inline systemtest/src/test/java/io/strimzi/systemtest/log/LogSettingST.java:276]: Is this needed?

File: systemtest/src/test/java/io/strimzi/systemtest/log/LogSettingST.java [hunk 3]
```diff
@@ -314,9 +309,8 @@ void testConnectLogSetting(ExtensionContext extensionContext) {
     @ParallelTest
     @Tag(MIRROR_MAKER)
     void testMirrorMakerLogSetting(ExtensionContext extensionContext) {
-        final TestStorage testStorage = storageMap.get(extensionContext);
-        String clusterName = testStorage.getClusterName();
-        String mirrorMakerName = clusterName + "-mirror-maker";
+        final TestStorage testStorage = new TestStorage(extensionContext);
+        String mirrorMakerName = testStorage.getClusterName() + "-mirror-maker";
```
Comment (im-konge) [inline systemtest/src/test/java/io/strimzi/systemtest/log/LogSettingST.java:313]: Same here

File: systemtest/src/test/java/io/strimzi/systemtest/log/LogSettingST.java [hunk 4]
```diff
@@ -368,31 +361,29 @@ void testMirrorMaker2LogSetting(ExtensionContext extensionContext) {
...
     }
 
     @ParallelTest
     @Tag(BRIDGE)
     void testBridgeLogSetting(ExtensionContext extensionContext) {
-        final TestStorage testStorage = storageMap.get(extensionContext);
-        final String clusterName = testStorage.getClusterName();
-        final String bridgeName = clusterName + "-bridge";
+        final TestStorage testStorage = new TestStorage(extensionContext);
+        final String bridgeName = testStorage.getClusterName() + "-bridge";
```
Comment (im-konge) [inline systemtest/src/test/java/io/strimzi/systemtest/log/LogSettingST.java:386]: Same here

File: systemtest/src/test/java/io/strimzi/systemtest/metrics/MetricsST.java [hunk 5]
```diff
@@ -296,7 +296,7 @@ void testKafkaConnectAndConnectorMetrics(ExtensionContext extensionContext) {
     @IsolatedTest
     @Tag(INTERNAL_CLIENTS_USED)
     void testKafkaExporterMetrics(ExtensionContext extensionContext) {
-        final TestStorage testStorage = storageMap.get(extensionContext);
+        final TestStorage testStorage = new TestStorage(extensionContext);
         final String producerName = "producer-" + new Random().nextInt(Integer.MAX_VALUE);
         final String consumerName = "consumer-" + new Random().nextInt(Integer.MAX_VALUE);
         final String kafkaStrimziPodSetName = KafkaResources.kafkaComponentName(kafkaClusterFirstName);
```
Comment (im-konge) [inline systemtest/src/test/java/io/strimzi/systemtest/metrics/MetricsST.java:302]: These will be removed in another PR or?

File: systemtest/src/test/java/io/strimzi/systemtest/metrics/MetricsST.java [hunk 6]
```diff
@@ -543,7 +543,7 @@ void testMirrorMaker2Metrics(ExtensionContext extensionContext) {
     @ParallelTest
     @Tag(BRIDGE)
     void testKafkaBridgeMetrics(ExtensionContext extensionContext) {
-        final TestStorage testStorage = storageMap.get(extensionContext);
+        final TestStorage testStorage = new TestStorage(extensionContext);
         String producerName = "bridge-producer";
         String consumerName = "bridge-consumer";
```
Comment (im-konge) [inline systemtest/src/test/java/io/strimzi/systemtest/metrics/MetricsST.java:548]: Same applies here

File: systemtest/src/test/java/io/strimzi/systemtest/operators/ReconciliationST.java [hunk 7]
```diff
@@ -68,134 +66,127 @@ public class ReconciliationST extends AbstractST {
     @Tag(CONNECT_COMPONENTS)
     @KRaftNotSupported("Probably bug - https://github.com/strimzi/strimzi-kafka-operator/issues/6862")
     void testPauseReconciliationInKafkaAndKafkaConnectWithConnector(ExtensionContext extensionContext) {
-        final TestStorage testStorage = storageMap.get(extensionContext);
-        final String namespaceName = StUtils.getNamespaceBasedOnRbac(Environment.TEST_SUITE_NAMESPACE, extensionContext);
-        final String clusterName = testStorage.getClusterName();
-        String kafkaSsName = KafkaResources.kafkaComponentName(clusterName);
+        final TestStorage testStorage = new TestStorage(extensionContext);
+        String kafkaSsName = KafkaResources.kafkaComponentName(testStorage.getClusterName());
```
Comment (im-konge) [inline systemtest/src/test/java/io/strimzi/systemtest/operators/ReconciliationST.java:70]: That's not needed or?

File: systemtest/src/test/java/io/strimzi/systemtest/operators/ReconciliationST.java [hunk 8]
```diff
@@ -68,134 +66,127 @@ public class ReconciliationST extends AbstractST {
...
 
-        final LabelSelector kafkaSelector = KafkaResource.getLabelSelector(clusterName, kafkaSsName);
+        final LabelSelector kafkaSelector = KafkaResource.getLabelSelector(testStorage.getClusterName(), kafkaSsName);
```
Comment (im-konge) [inline systemtest/src/test/java/io/strimzi/systemtest/operators/ReconciliationST.java:72]: This one as well

File: systemtest/src/test/java/io/strimzi/systemtest/operators/topic/ThrottlingQuotaST.java [hunk 9]
```diff
@@ -55,7 +55,7 @@ public class ThrottlingQuotaST extends AbstractST {
     @KRaftWithoutUTONotSupported
     @UTONotSupported("https://github.com/strimzi/strimzi-kafka-operator/issues/8864")
     void testThrottlingQuotasDuringAllTopicOperations(ExtensionContext extensionContext) {
-        final TestStorage testStorage = new TestStorage(extensionContext, Environment.TEST_SUITE_NAMESPACE);
+        final TestStorage testStorage = new TestStorage(extensionContext);
```
Comment (im-konge) [inline systemtest/src/test/java/io/strimzi/systemtest/operators/topic/ThrottlingQuotaST.java:58]: Why did you delete it?

File: systemtest/src/test/java/io/strimzi/systemtest/rollingupdate/RollingUpdateST.java [hunk 10]
```diff
@@ -600,25 +596,22 @@ void testBrokerConfigurationChangeTriggerRollingUpdate(ExtensionContext extensio
...
 
-        resourceManager.createResourceWithWait(extensionContext, KafkaTemplates.kafkaPersistent(clusterName, 3, 3)
+        resourceManager.createResourceWithWait(extensionContext, KafkaTemplates.kafkaPersistent(testStorage.getClusterName(), 3, 3)
             .editMetadata()
                 .withNamespace(Environment.TEST_SUITE_NAMESPACE)
             .endMetadata()
             .build());
 
-        Map<String, String> kafkaPods = PodUtils.podSnapshot(Environment.TEST_SUITE_NAMESPACE, kafkaSelector);
+        Map<String, String> kafkaPods = PodUtils.podSnapshot(Environment.TEST_SUITE_NAMESPACE, testStorage.getKafkaSelector());
```
Comment (im-konge) [inline systemtest/src/test/java/io/strimzi/systemtest/rollingupdate/RollingUpdateST.java:607]: Instead of `Environment.TEST_SUITE_NAMESPACE` you can use `testStorage.getNamespaceName()`, right?

**Review by im-konge — COMMENTED** [review]
Thanks for the PR, few nits

File: systemtest/src/test/java/io/strimzi/systemtest/operators/topic/ThrottlingQuotaST.java (hunk 9 above)
Comment (im-konge) [reply systemtest/src/test/java/io/strimzi/systemtest/operators/topic/ThrottlingQuotaST.java:58]: Never mind, I see why you deleted it here.

**Comment by jankalinic** [comment]
The only issue in azp regression is with testKafkaConnectAndConnectorStatus => missing topic `example-topic-name` in connect, this isn't however related to the PR, as it does not change code which might result in such error. Re-tested locally several times and the topic is present.
